src/HotChocolate/Marten/src/Data/MartenExpressionTranslator.cs

```diff
@@ -0,0 +1,41 @@
+using System.Linq.Expressions;
+
+namespace HotChocolate.Data.Marten;
+
+public static class MartenExpressionTranslator
+{
+    public static Expression TranslateFilterExpression(Expression filterExpression)
+    {
+        if (filterExpression is BinaryExpression
+            {
+                NodeType: ExpressionType.AndAlso,
+                Left: BinaryExpression leftBinaryExpression,
+                Right: { } rightExpression
+            })
+        {
+            if (ReferenceComparisonExpressionShouldBeRemoved(leftBinaryExpression))
+            {
+                return rightExpression;
+            }
+        }
+        return filterExpression;
+    }
+
+    private static bool ReferenceComparisonExpressionShouldBeRemoved(BinaryExpression expression)
+    {
+        return expression.NodeType == ExpressionType.NotEqual
```

Comment:
Yes, but only if it's on the base object of the query, and not a member access expression.
So:
`where([x] != null AndAlso [x].Rating > 50)` will get simplified to `where([x].Rating > 50)`
But:
`where([x].NestedObject  != null AndAlso [x].NestedObject.Rating > 50)` will get not get simplified.

Reply:
Also, I don't rewrite any of the expressions. For a binary expression like `where([x] != null AndAlso [x].Rating > 50)`, to simplify, I simply ignore the left operand and return the right one.

Reply:
What exactly does `context.InMemory` indicate? And no. Actually Marten will work just fine without the null checks, so we don't really need them.

Reply:
@jabellard `context.InMemory` removes all the null checks. We do this for ef core to optimize the expressions. Do you think this could also work for marten?

Reply:
@PascalSenn Ok. I see. Unlike EF, Marten does not support an in memory database, but `context.InMemory` evaluates to `true` when working with the Marten queryable for some reason so the null checks are added. However, given that Marten will work fine without the null checks, everything should work like a charm if we just get rid of the null checks at the following sites:
https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Data/src/Data/Filters/Expressions/Handlers/QueryableDefaultFieldHandler.cs#L112
https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Data/src/Data/Sorting/Expressions/Handlers/QueryableDefaultSortFieldHandler.cs#L73

What do you recommend as the best, least intrusive way to go about doing that for Marten?

Reply:
@PascalSenn , Never mind. Figured it out. Can you take a look?